src/index.ts

```diff
@@ -1,96 +1,32 @@
+import { openAttestationDnsTxtIdentity } from "./verifiers/openAttestationDnsTxtIdentity";
+import { openAttestationEthereumDocumentStoreIssued } from "./verifiers/openAttestationEthereumDocumentStoreIssued";
+import { openAttestationEthereumDocumentStoreRevoked } from "./verifiers/openAttestationEthereumDocumentStoreRevoked";
+
+const defaultVerifiers: Verifier[] = [
+  openAttestationTamperCheck,
+  openAttestationEthereumDocumentStoreIssued,
+  openAttestationEthereumDocumentStoreRevoked,
+  openAttestationDnsTxtIdentity
 ];
 
-const getAllChecks = (
-  document: SignedDocument,
-  smartContracts: OpenAttestationContract[]
-): VerificationChecks => [
-  verifyHash(document),
-  verifyIssued(document, smartContracts),
-  verifyRevoked(document, smartContracts)
-];
-
-const isDocumentValid = (
-  hash: ResolveType<typeof verifyHash>,
-  issued: ResolveType<typeof verifyIssued>,
-  revoked: ResolveType<typeof verifyRevoked>
-) => hash.checksumMatch && issued.issuedOnAll && !revoked.revokedOnAny;
-
-/**
- * @param  {object} document Entire document object to be validated
- * @param  {string} network Network to check against, defaults to "homestead". Other valid choices: "ropsten", "kovan", etc
- * @returns
- */
-export const verify = async (
-  document: SignedDocument,
-  network = "homestead"
-) => {
-  const smartContracts = documentToSmartContracts(document, network);
-  const checks = getAllChecks(document, smartContracts);
-  const [hash, issued, revoked] = await Promise.all(checks);
-
-  return {
-    hash,
-    issued,
-    revoked,
-    valid: isDocumentValid(hash, issued, revoked)
-  };
-};
-
-/**
- * @param  {object} document Entire document object to be validated
- * @param  {string} network Network to check against, defaults to "homestead". Other valid choices: "ropsten", "kovan", etc
- * @returns  {array} Array of promises, each promise corresponds to a verification check.
- *                   The last promise resolves to the overall validity based on all the checks.
- */
-export const verifyWithIndividualChecks = (
-  document: SignedDocument,
-  network = "homestead"
-): VerificationChecksWithValidity => {
-  const smartContracts = documentToSmartContracts(document, network);
-  const [hash, issued, revoked] = getAllChecks(document, smartContracts);
-
-  // If any of the checks are invalid, resolve the overall validity early
-  const valid = Promise.all([
-    new Promise(async (resolve, reject) =>
-      (await hash).checksumMatch ? resolve() : reject()
-    ),
-    new Promise(async (resolve, reject) =>
-      (await issued).issuedOnAll ? resolve() : reject()
-    ),
-    new Promise(async (resolve, reject) =>
-      (await revoked).revokedOnAny ? reject() : resolve()
-    )
-  ])
-    .then(() => true)
-    .catch(() => false);
-
-  return [hash, issued, revoked, valid];
+const defaultVerificationManager = verificationManager(defaultVerifiers);
+const isValid = (verificationFragments: VerificationFragment[]) =>
```

Comment:
isValid(document, requiredChecksClasses)

usage example
isValid(document, ["ISSUER_IDENTITY", "ISSUANCE_STATUS", "INTEGRITY"])

Reply:
Basically isValid checks the fragments that EVERY required checks classes to have at EVERY test either "VALID" or "SKIPPED" and has at least one "VALID".

For example for a documentStore
issued: valid
revoked: invalid
will result in "invalid"

In another case: 
issued: valid
revoked: skipped
will result in "valid"

Reply:
done